Added instructions to update kubeconfig after manual certs renewal or cluster upgrade (#7890)
aws/eks-anywhere: docs/content/en/docs/clustermgmt/security/manually-renew-certs.md

5. Repeat the above steps for all control plane nodes.

You can similarly use the above steps to rotate a single certificate instead of all certificates.

> #### Update kubeconfig on Admin machine

Reviewer: ```suggestion
#### Update your kubeconfig certificate
```

Admin machines are ephemeral. Suggesting they need to update the admin machine is a little misleading as it may no longer exist.

> Post completion of manual certificate rotation or if your Cluster was created more than a year ago, your kubeconfig file on Admin machine will have outdated certificates and would result in following error,

Reviewer: ```suggestion
Your local kubeconfig used to interact with the cluster contains a certificate that expires after 1 year. When you rotate cluster certificates a new kubeconfig with a new certificate is created as a Secret in the cluster. If you do not retrieve the new kubeconfig and your local kubeconfig certificate expires you will receive the following error:
```

```
Error: Couldn't get current Server API group list: the server has asked for the client to provide credentials error: you must be logged in to the server.
This error typically occurs when the cluster certificates have been renewed or extended during the upgrade process. To resolve this issue, you need to update your local kubeconfig file with the new cluster credentials.
```

> Follow below steps to update kubeconfig on Admin machine.

Reviewer: ```suggestion
To extract your new kubeconfig from the cluster do the following.
```

> 1. SSH to one of the Control Plane nodes and run the following command to validate connection with API Server, export kubeconfig from `${CLUSTER_NAME}-kubeconfig` secret object (`eksa-system` namespace) using kubectl and copy kubeconfig file to `/tmp` directory.

Reviewer: This step is only necessary if the local kubeconfig has already expired. If not, they should be able to retrieve the secret with the existing cert data.

Can we move the step to retrieve the admin.conf to a separate subheading "What do I do if my local kubeconfig has expired?"

```
ssh <YOUR_CONTROLPLANE_IP>
```

{{< tabpane >}}
{{< tab header="Ubuntu or RHEL" lang="bash" >}}

export CLUSTER_NAME="<YOUR_CLUSTER_NAME_HERE>"

cat /var/lib/kubeadm/admin.conf
export KUBECONFIG="/var/lib/kubeadm/admin.conf"

> kubectl get nodes -o wide

Reviewer: These steps aren't necessary so I think we can eliminate them.

kubectl get secrets -A

kubectl get secret ${CLUSTER_NAME}-kubeconfig -n eksa-system -o yaml > new-admin.kubeconfig

> cat new-admin.kubeconfig | base64 -d > /tmp/new-admin-decoded.kubeconfig

Reviewer: You shouldn't need to decode the admin.conf. Decoding is only needed for the secret we extract from the cluster.

{{< /tab >}}

{{< tab header="Bottlerocket" lang="bash" >}}
# you would be in the admin container when you ssh to the Bottlerocket machine
# open a root shell
sudo sheltie

> export CLUSTER_NAME="<YOUR_CLUSTER_NAME_HERE>"

Reviewer: Unused.

cat /var/lib/kubeadm/admin.conf
export KUBECONFIG="/var/lib/kubeadm/admin.conf"

kubectl get nodes -o wide

> kubectl get secrets -A

Reviewer: Ditto.

cat new-admin.kubeconfig | base64 -d > /run/host-containerd/io.containerd.runtime.v2.task/default/admin/rootfs/tmp/new-admin-decoded.kubeconfig

{{< /tab >}}
{{< /tabpane >}}

> 2. **SSH to Admin Machine**, download the kubeconfig file from ControlPlane to your Admin machine and access Kubernetes Cluster

Reviewer: ```suggestion
2. From your admin machine, download the kubeconfig file from the ControlPlane node and use it to access your Kubernetes Cluster.
```

```
ssh <ADMIN_MACHINE_IP>

export CONTROLPLANE_IP=""
scp -i <keypair>@${CONTROLPLANE_IP}:/tmp/new-admin-decoded.kubeconfig .

# OR SFTP

> sftp -i <keypair>@${CONTROLPLANE_IP}:/tmp/new-admin-decoded.kubeconfig .

Reviewer: Why would scp not work? I think we can include a single approach and let the user decide how they want to do it (I can think of several other approaches and we don't want to enumerate them).

```
Error: Couldn't get current Server API group list: the server has asked for the client to provide credentials error: you must be logged in to the server.
This error typically occurs when the cluster certificates have been renewed or extended during the upgrade process. To resolve this issue, you need to update your local kubeconfig file with the new cluster credentials.
```

You can extract your new kubeconfig using the following steps.

1. SSH to one of the Control Plane nodes and run the following command to validate connection with API Server, export kubeconfig from `${CLUSTER_NAME}-kubeconfig` secret object (`eksa-system` namespace) using kubectl and copy kubeconfig file to `/tmp` directory.

```
ssh <YOUR_CONTROLPLANE_IP>
```

{{< tabpane >}}
{{< tab header="Ubuntu or RHEL" lang="bash" >}}

export CLUSTER_NAME="<YOUR_CLUSTER_NAME_HERE>"

cat /var/lib/kubeadm/admin.conf
export KUBECONFIG="/var/lib/kubeadm/admin.conf"

kubectl get secret ${CLUSTER_NAME}-kubeconfig -n eksa-system -o yaml > new-admin.kubeconfig

cat new-admin.kubeconfig > /tmp/new-admin-decoded.kubeconfig

{{< /tab >}}

{{< tab header="Bottlerocket" lang="bash" >}}

> # you would be in the admin container when you ssh to the Bottlerocket machine

Reviewer 2: ```suggestion
# You would need to be in the admin container when you ssh to the Bottlerocket machine
```

```
Error: Couldn't get current Server API group list: the server has asked for the client to provide credentials error: you must be logged in to the server.
This error typically occurs when the cluster certificates have been renewed or extended during the upgrade process. To resolve this issue, you need to update your local kubeconfig file with the new cluster credentials.
```

You can extract your new kubeconfig using the following steps.

1. SSH to one of the Control Plane nodes and run the following command to validate connection with API Server, export kubeconfig from `${CLUSTER_NAME}-kubeconfig` secret object (`eksa-system` namespace) using kubectl and copy kubeconfig file to `/tmp` directory.

```
ssh -i <YOUR_PRIVATE_KEY> <USER_NAME>@<YOUR_CONTROLPLANE_IP> # USER_NAME should be ec2-user for bottlerocket, ubuntu for Ubuntu ControlPlane machine Operating System

```

{{< tabpane >}}
{{< tab header="Ubuntu or RHEL" lang="bash" >}}

export CLUSTER_NAME="<YOUR_CLUSTER_NAME_HERE>"

cat /var/lib/kubeadm/admin.conf

> export KUBECONFIG="/var/lib/kubeadm/admin.conf"

Reviewer 3: That's fine. I just didn't know where it was used.

kubectl get secret ${CLUSTER_NAME}-kubeconfig -n eksa-system -o yaml > new-admin.kubeconfig

cat new-admin.kubeconfig > /tmp/new-admin-decoded.kubeconfig

{{< /tab >}}

{{< tab header="Bottlerocket" lang="bash" >}}

# You would need to be in the admin container when you ssh to the Bottlerocket machine
# open a root shell
sudo sheltie

cat /var/lib/kubeadm/admin.conf
export KUBECONFIG="/var/lib/kubeadm/admin.conf"

cat /var/lib/kubeadm/admin.conf > /run/host-containerd/io.containerd.runtime.v2.task/default/admin/rootfs/tmp/new-admin.kubeconfig

{{< /tab >}}
{{< /tabpane >}}

2. From your admin machine, download the kubeconfig file from the ControlPlane node and use it to access your Kubernetes Cluster.

```
ssh <ADMIN_MACHINE_IP>

export CONTROLPLANE_IP=""
scp -i <keypair>@${CONTROLPLANE_IP}:/tmp/new-admin.kubeconfig .

> ls -ltr 

Reviewer 3: Okay

      - https://xxx.xxx.xxx.xxx:2379
      - https://xxx.xxx.xxx.xxx:2379
```

### What do I do if my local kubeconfig has expired?

> Your local kubeconfig, used to interact with the cluster contains a certificate that expires after 1 year. When you rotate cluster certificates, a new kubeconfig with a new certificate is created as a Secret in the cluster. If you do not retrieve the new kubeconfig and your local kubeconfig certificate expires, you will receive the following error:

Reviewer 3: ```suggestion
Your local kubeconfig, used to interact with the cluster, contains a certificate that expires after 1 year. When you rotate cluster certificates, a new kubeconfig with a new certificate is created as a Secret in the cluster. If you do not retrieve the new kubeconfig and your local kubeconfig certificate expires, you will receive the following error:
```
